solr/core/src/java/org/apache/solr/update/LeafSorter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.update;
+
+import java.lang.invoke.MethodHandles;
+import java.util.Comparator;
+import org.apache.lucene.index.LeafReader;
+import org.apache.lucene.index.SegmentReader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public interface LeafSorter {
+
+  Comparator<LeafReader> getLeafSorter();
```

Comment:
looks weird,  LeafSorter has a getLeafSorter that does not actually return a LeafSorter?
Maybe the interface should be renamed LeafSorterSupplier

Reply:
renamed.